sky/resources.py

```diff
@@ -572,14 +601,18 @@ def less_demanding_than(self,
         # self <= other
         return True
 
-    def is_launchable_fuzzy_equal(self, other: 'Resources') -> bool:
-        """Whether the resources are the fuzzily same launchable resources."""
-        assert self.cloud is not None and other.cloud is not None
-        if not self.cloud.is_same_cloud(other.cloud):
+    def should_be_blocked_by(self, blocked: 'Resources') -> bool:
+        """Is this Resources fuzzily equal to the blocked Resources?"""
+        assert self.cloud is not None and blocked.cloud is not None
+        if not self.cloud.is_same_cloud(blocked.cloud):
             return False
-        if self._instance_type is not None or other.instance_type is not None:
-            return self._instance_type == other.instance_type
-        return self.accelerators.keys() == other.accelerators.keys()
+        if self._instance_type is not None or blocked.instance_type is not None:
+            return self._instance_type == blocked.instance_type
```

Comment:
Doesn't this block the resources in different regions?

`sky launch -c test-a100 --gpus A100:8 --cloud aws` fails after trying only 1 region.

Reply:
A proposed fix:
```
    def should_be_blocked_by(self, blocked: 'Resources') -> bool:
        """Whether this Resources matches the blocked Resources.

        If a field in `blocked` is None, it should be considered as a wildcard.
        """
        is_matched = True
        if blocked.cloud is not None and not self.cloud.is_same_cloud(blocked.cloud):
            is_matched = False
        if blocked.instance_type is not None and self.instance_type != blocked.instance_type:
            is_matched = False
        if blocked.region is not None and self._region != blocked.region:
            is_matched = False
        if blocked.zone is not None and self._zone != blocked.zone:
            is_matched = False
        if blocked.accelerators is not None and self.accelerators != blocked.accelerators:
            is_matched = False
        return is_matched
```

Reply:
Great catch! Thanks for pointing this out. I made mistake when addressing one of the previous reviews.

Reply:
I think `If a field in `blocked` is None, it should be considered as a wildcard.` is a little bit confusing. To me, if the blocked resource has a "wildcard" field, it sounds like any resource matches the blocked resource in terms of that field. In the code, it's opposite, right?

Therefore, I'd like to just skip that line in the comment. The code itself looks pretty clear to me. WDYT?

Reply:
I think in the code, that means any value in the field of `resources` matches the wildcard. Can we still keep that comment by changing it to 
```
If a field in `blocked` is None, it should be considered as a wildcard for that field.
```

Reply:
Oh I misunderstood the code here. Yes, it works as a wildcard. I will add that comment.